lib/components/molecules/Carousel/Carousel.js

```diff
@@ -0,0 +1,398 @@
+// @flow
+/**
+ * Carousel
+ */
+import React, { useContext, useState } from 'react';
+import styled from 'styled-components';
+import classnames from 'classnames';
+
+import stylesCarousel from './Carousel.style';
+import { Props, AnimationKeys } from './types';
+import { ListVanilla } from '../List';
+import Button from '../../atoms/Button';
+import useInterval from './useInterval.hook';
+
+const CarouselContext = React.createContext([]);
+
+/**
+ * Add unique Index to all slides.
+ *
+ * @param {Array} slides
+ *   all slides
+ *
+ * @return All slides with added slideIndex key as prop.
+ */
+const slideWithIndexKey = slides => {
+  let i = 0;
+  const children = React.Children.map(slides, slide => {
+    i += 1;
+    return React.cloneElement(slide, {
```

Comment:
This is unnecessary complication. No need to do this if you want to set a incremental id, use a method that returns this value in the default Props.